app/src/org/commcare/views/widgets/SelectOneWidget.java

```diff
@@ -101,17 +107,45 @@ public SelectOneWidget(Context context, FormEntryPrompt prompt) {
                     addView(divider);
                 }
             }
+            addClearButton(context, s != null && !prompt.isReadOnly());
         }
     }
 
+    private void addClearButton(Context context, boolean show) {
+        clearButton = new Button(context);
```

Comment:
would be good to abstract this code into xml. We do have a ton of layout code in Java but it's preferrable to not add onto that mess.